drivers/flash/Kconfig.stm32_ospi

```diff
@@ -3,8 +3,8 @@
 # Copyright (c) 2022 STMicroelectronics
 # SPDX-License-Identifier: Apache-2.0
 
-DT_STM32_OCTOSPI_1_HAS_DMA := $(dt_nodelabel_has_prop,octospi1,dmas)
-DT_STM32_OCTOSPI_2_HAS_DMA := $(dt_nodelabel_has_prop,octospi2,dmas)
+DT_STM32_OCTOSPI_1_HAS_DMA := $(dt_compat_any_has_prop,octospi1,dmas)
+DT_STM32_OCTOSPI_2_HAS_DMA := $(dt_compat_any_has_prop,octospi2,dmas)
```

Comment:
Aim of `dt_compat_any_has_prop()` is to check if there is at least one node with a given compat that has a given prop.
So, it could be replaced by :
`DT_STM32_OCTOSPI_HAS_DMA := $(dt_compat_any_has_prop,<compat>,dmas)`